### ISSUE TITLE
sync_folders action doesn't work with Active Directory domain\username

### ISSUE THREAD

**Issue by chrisbaldauf** [tknerr/vagrant-managed-servers#38]
We're using [Likewise Open](https://help.ubuntu.com/community/LikewiseOpen) as a mechanism to authenticate users on Linux boxes. When a user logs in to a server, they're required to enter their username in `domain\username` format. This hits a snag when entered from a Windows machine (haven't tested Mac yet). 

```
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

chown -R domain\cbaldauf '/vagrant'

Stdout from the command:

Stderr from the command:

stdin: is not a tty
chown: invalid user: 'domaincbaldauf'
```

By patching synch_folders.rb as follows appears to correct the issue for us and I believe that it shouldn't hurt anyone using non-domain authentication.

```
            # Create the guest path
            env[:machine].communicate.sudo("mkdir -p '#{guestpath}'")
+            ssh_info[:username].gsub!("\\","\\\\\\\\")
            env[:machine].communicate.sudo(
              "chown -R #{ssh_info[:username]} '#{guestpath}'")

            # Rsync over to the guest path using the SSH info
            ssh_key_options = Array(ssh_info[:private_key_path]).map { |path| "-i '#{path}' " }.join
            ssh_proxy_commands = Array(ssh_info[:proxy_command]).map { |command| "-o ProxyCommand='#{command}' "}.join
            command = [
              "rsync", "--verbose", "--archive", "-z",
              "--exclude", ".vagrant/", "--exclude", "Vagrantfile",
              "-e", "ssh -p #{ssh_info[:port]} -o StrictHostKeyChecking=no #{ssh_key_options} #{ssh_proxy_commands}",
              hostpath,
              "#{ssh_info[:username]}@#{ssh_info[:host]}:#{guestpath}"]
```

This escapes the backslash in both the chown and rsync command and the provisioning run completed successfully after such a patch. I'll open a PR, so let me know if you can think of any concerns that you may have with this change.

**Comment by chrisbaldauf** [comment]
I tested the behavior on my Mac and confirmed that it displays the error message from a Mac host as well as long as the username contains a backslash. 

```
==> server1.mydomain.com: Linking with managed server server1.mydomain.com
==> server1.mydomain.com:  -- Server: server1.mydomain.com
    server1.mydomain.com: 
    server1.mydomain.com: Inserting generated public key within guest...
    server1.mydomain.com: Removing insecure key from the guest if its present...
    server1.mydomain.com: Key inserted! Disconnecting and reconnecting using new SSH key...
==> server1.mydomain.com: Rsyncing folder: /Users/cbaldauf/dev/github/chrisbaldauf/vagrant-orchestrate/ => /vagrant
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

chown -R mydomain\cbaldauf '/vagrant'

Stdout from the command:

Stderr from the command:

stdin: is not a tty
chown: invalid user: 'mydomaincbaldauf'
```

Then it got weirder. The patch as proposed breaks auth on Mac. I wound up having to escape the username in the `chown` command but not in the rsync command. I think I want to take a step back, think this through a bit more and do some additional testing. In all likelyhood, I'll look at this some more on Monday.

Thanks for the feedback.

**Comment by tknerr** [comment]
fixed via #39 by quoting the `'user\name'`

**Comment by tknerr** [comment]
released as v0.6.1
